File: android/app/src/main/java/fr/gshz/hideandseek/domain/model/QuestionCategory.kt
package fr.gshz.hideandseek.domain.model

enum class QuestionCategory(val wireValue: String) {
    Radar("radar"),
    Thermometer("thermometer"),
    Matching("matching"),
    Measuring("measuring"),
    Tentacles("tentacles"),
    Photos("photos"),
    ;

    companion object {
        fun fromWireValue(value: String): QuestionCategory? = entries.firstOrNull { it.wireValue == value }
    }
}

File: backend/src/Enum/QuestionCategory.php
<?php

declare(strict_types=1);

namespace App\Enum;

enum QuestionCategory: string
{
    case Radar = 'radar';
    case Thermometer = 'thermometer';
    case Matching = 'matching';
    case Measuring = 'measuring';
    case Tentacles = 'tentacles';
    case Photos = 'photos';

    public function requiresFeature(): bool
    {
        return match ($this) {
            self::Matching, self::Measuring, self::Tentacles => true,
            self::Radar, self::Thermometer, self::Photos => false,
        };
    }
}
